server/cli/src/backup/backup.rs

```diff
@@ -0,0 +1,151 @@
+use super::*;
+use chrono::Utc;
+use copy_dir::copy_dir;
+use service::settings::{BackupSettings, Settings};
+use std::{fs, io, path::PathBuf, process::Command, str::FromStr};
+
+pub(crate) fn backup(settings: &Settings) -> Result<(), BackupError> {
+    let Some(BackupSettings {
+        backup_dir,
+        pg_bin_dir,
+    }) = settings.backup.clone()
+    else {
+        return Err(BackupError::BackupConfigurationMissing);
+    };
+
+    let Dirs {
+        backup_name,
+        file_dir,
+        database_dir,
+    } = create_backup_dir(backup_dir)?;
+
+    copy_files(settings, &file_dir)?;
+
+    // Backup database
+    if cfg!(feature = "postgres") {
+        dump_postgres_database(settings, &database_dir, pg_bin_dir)?;
+    } else {
+        copy_sqlite_files(settings, &database_dir)?;
+    }
+
+    println!("{backup_name}");
```

Comment:
I actually wanted just the name of backup, to make it easy to pipe or copy to clipboard with cli, added comment here, maybe that would help explain ? (otherwise I can make a flag to print `Backup completed ..` or just the backup folder name).
c170a329cc8ff94e0f70823268b13f660cf74bcc

Here is the commit and this is what it looks like:
![Screenshot 2024-08-23 at 3 17 20 PM](https://github.com/user-attachments/assets/b6110bb7-a87d-4234-915a-0726f60bc430)

Reply:
nice. in which case a batch / shell script example of catching the backup name and return value to indicate success would be pretty sweet to add to the readme.

Was just testing the return values like so, and it's all good 🙏 

```
#!/bin/bash

BACKUP_FOLDER=$(./target/debug/remote_server_cli backup)

RETURN=$?

if [ $RETURN -eq 0 ];
then
  echo "Backup completed successfully in folder $BACKUP_FOLDER"
  exit 0
else
  echo "Backup failed!"
  exit $RETURN
fi
```